src/components/floatingButton/ArrowUpFloatingButton.tsx

```diff
@@ -0,0 +1,56 @@
+'use client';
+
+/**
+ TODO
+ - [x] 클릭 시 최상단으로 이동 구현
+ - [x] 스크롤에 따른 버튼 디자인 구현
+ - [x] 다른 페이지에서도 사용할 수 있도록 공통 컴포넌트화(리팩토링)
+ */
+
+import { useEffect, useState } from 'react';
+import { assignInlineVars } from '@vanilla-extract/dynamic';
+
+import * as styles from './FloatingContainer.css';
+import ArrowUpIcon from '/public/icons/arrow_up.svg';
+
+export default function ArrowUpFloatingButton() {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const handleScrollToTop = () => {
+    if (!isVisible) return;
+
+    window.scrollTo({
+      top: 0,
+      behavior: 'smooth',
+    });
+  };
+
+  const visibleButton = () => {
+    if (window.scrollY < 500) {
+      setIsVisible(false);
+    } else {
+      setIsVisible(true);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('scroll', visibleButton);
+
+    return () => {
+      window.removeEventListener('scroll', visibleButton);
+    };
+  }, []);
+
+  return (
+    <div
+      className={styles.arrowUpButton}
+      onClick={handleScrollToTop}
+      style={assignInlineVars({
+        [styles.opacitySize]: isVisible ? '1' : '0',
+        [styles.cursor]: isVisible ? 'pointer' : 'default',
+      })}
+    >
+      <ArrowUpIcon alt="상단으로 이동하기 버튼" className={styles.icon} />
+    </div>
```

Comment:
아침 데일리스크럼 때 제안드린 부분 구현 방식을 고민해봤습니다!

```suggestion
//아예 없애고 노출시키는 방향
    <>
      {isVisible && (
        <div className={styles.arrowUpButton} onClick={handleScrollToTop}>
          <ArrowUpIcon alt="상단으로 이동하기 버튼" className={styles.icon} />
        </div>
      )}
    </>
```

``` 
//CSS는 transition 대신 애니메이션으로
    const fadeIn = keyframes({
    '0%': { opacity: 0 },
    '100%': { opacity: 1 },
    });

    export const arrowUpButton = style([
    addButton,
    {
        animation: `${fadeIn} 300ms ease`,
    },
    ]);
```